relay-event-normalization/src/normalize/span/ai.rs

```diff
@@ -364,4 +378,148 @@ mod tests {
         }
         ");
     }
+
+    /// Test that the AI operation type is inferred from a gen_ai.operation.name attribute.
+    #[test]
+    fn test_infer_ai_operation_type_from_gen_ai_operation_name() {
+        let mut operation_types = HashMap::new();
+        operation_types.insert(Pattern::new("*").unwrap(), "ai_client".to_owned());
+        operation_types.insert(Pattern::new("invoke_agent").unwrap(), "agent".to_owned());
+        operation_types.insert(
+            Pattern::new("gen_ai.invoke_agent").unwrap(),
+            "agent".to_owned(),
+        );
+
+        let operation_type_map = AiOperationTypeMap {
+            version: 1,
+            operation_types,
+        };
+
+        let span = r#"{
+            "data": {
+                "gen_ai.operation.name": "invoke_agent"
+            }
+        }"#;
+        let mut span: Span = Annotated::from_json(span).unwrap().into_value().unwrap();
+        infer_ai_operation_type(&mut span, &operation_type_map);
+        assert_eq!(
+            span.data
+                .value()
+                .unwrap()
+                .gen_ai_operation_type
+                .value()
+                .unwrap()
+                .as_str(),
+            "agent"
+        );
```

Comment:
The `get_value` macro was made to access deeply nested `Annotated` fields, also potentially suggest looking into the `assert_annotated_snapshot`/`assert_json_snapshot` macros if you want to assert multiple fields.

```suggestion
        let span = r#"{
            "data": {
                "gen_ai.operation.name": "invoke_agent"
            }
        }"#;
        let mut span = Annotated::from_json(span).unwrap();
        infer_ai_operation_type(span.value_mut().as_mut().unwrap(), &operation_type_map);
        assert_eq!(
            get_value!(span.data.gen_ai_operation_name!).as_str(),
            "agent"
        );
```